src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -1500,20 +1488,60 @@ public GHBranchSync sync(String branch) throws IOException {
      *
      * @param org
      *            the org
-     * @return Newly forked repository that belong to you.
+     * @return Newly forked repository that belongs to you.
      * @throws IOException
      *             the io exception
+     * @deprecated Use {@link #createFork(String, String, boolean)} instead
      */
+    @Deprecated(forRemoval = true)
     public GHRepository forkTo(GHOrganization org) throws IOException {
-        root().createRequest()
-                .method("POST")
-                .with("organization", org.getLogin())
-                .withUrlPath(getApiTailUrl("forks"))
-                .send();
+        return createFork(org.getLogin(), null, false);
+    }
+
+    /**
+     * Creates a fork of this repository with optional parameters.
+     *
+     * @param organization
+     *            the organization to fork to, or null to fork to the authenticated user's account
+     * @param name
+     *            the name of the new repository, or null to use the same name as the original repository
+     * @param defaultBranchOnly
+     *            whether to fork only the default branch
+     * @return the newly forked repository
+     * @throws IOException
+     *             if an I/O error occurs
+     */
+    public GHRepository createFork(@Nullable String organization, @Nullable String name, boolean defaultBranchOnly)
+            throws IOException {
```

Comment:
You did basically exactly what I ask for with the deprecations, thanks! 
Did you leave out the `public GHRepository createFork()` intentionally or just WIP?  
If you still plan to provide the method overload with few parameters, the right signature is `public GHRepository createFork(boolean defaultBranchOnly)` instead of no parameters.  What are your thoughts?

Reply:
WiP, definitely. I'll work on it, thanks!